### PR TITLE
Add forward-looking dependency on arraycontext

### PR DESCRIPTION
**Pull request by inducer**
This isn't technically needed yet (`arraycontext` is nothing but an empty shell for now), but it'll help me with some [upstream PR headaches](https://github.com/inducer/meshmode/pull/181).

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ def main():
               "pytest>=2.3",
               "pytools>=2018.5.2",
               "modepy>=2013.3",
+              "arraycontext=2021.1",
```
Comment (matthiasdiener) [inline setup.py:49]: Should this be >= ?

Comment (inducer) [inline setup.py:49]:
```suggestion
              "arraycontext>=2021.1",
```

Comment (inducer) [reply setup.py]: Thanks, good catch! 5d686b6

**Review by matthiasdiener — APPROVED** [review]

**Comment by matthiasdiener** [comment]
Doc build will fail until an arraycontext package is released: https://readthedocs.org/projects/mirgecom/builds/13751500/

**Comment by inducer** [comment]
Yikes, sorry. #340.

**Pull request merged**